### ISSUE TITLE
Brave Leo - Lack of Accessibility for Screen Readers to Edit and Delete Individual Messages

### ISSUE THREAD

**Issue by aiyumi** [brave/brave-browser#51683]
### Description

Using the Orca screen reader on Linux, I can't find any way to edit or delete individual messages in Brave Leo. I even thought there weren't options to edit and delete individual messages, but I asked Leo and it answered it's possible. It told me that hovering the mouse over a message should bring a context menu with a pencil icon to edit the message and a trash can icon to delete the message. But as a blind, screen reader user, I rely on the keyboard and can't use the mouse. I asked Leo whether those actions are possible using only the keyboard, and it answered that I should focus on the message I want to edit or delete using the **Tab** key and press either **Enter** or **Spacebar** for the context menu to appear, navigate through the options using the arrow keys and activate them using **Enter**. But the **Tab** key doesn't focus on any message, and the screen reader doesn't give me any indication of the existence of a context menu with options to edit or delete messages.

### Steps to reproduce

1. On Linux, with the Orca screen reader turned on, open Brave and open Leo.
2. Type a question and press **Enter** to send it.
3. Use **Tab**/**Shift+Tab** to navigate through the page and try to focus on the question or Leo's response.

### Actual result

- Focus doesn't stop on the messages and jumps to the buttons near them. The screen reader announces the buttons (though not their labels. It only says "Button").
- The screen reader doesn't announce nor gives any indication that options to edit or delete messages exist.

### Expected result

According to what Leo told me that should happen:
- 
- When navigating using the **Tab**/**Shift+Tab** keys, focus should stop on the messages.
- Once a message is focused, pressing either **Enter** or **Spacebar** should bring a context menu with the option to edit and delete the focused message.
- Navigating through the menu with the **up** and **down** arrow keys should cycle through the options, and the screen reader should announce the focused option.
- Pressing **Enter** or **Spacebar** should activate the focused option.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

- Brave: 1.85.118 Chromium: 143.0.7499.169
- Revision: 1536f1c9cb01d4662751570659121a2a2bc0
- OS: Linux

### Channel information

- [x] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

- Screen reader: Orca 49.5
